lib/cdo/script_constants.rb

```diff
@@ -275,28 +271,26 @@ def self.teacher_dashboard_name(script)
 
   # Sections can be assigned to both courses and scripts. We want to make sure
   # we give teacher dashboard the same information for both sets of assignables.
-  # We also expect to be in a mixed world for a time where this info is asked for
-  # both by dashboard and pegasus, and we want to keep that in sync. We accomplish
-  # most of that through this shared method, leaving it to dashboard/pegasus
-  # to take care of translating name/cateogry (with the unfortunate effect that
-  # we could have two different translations).
-  # @param course_or_script [Course|Script] A row object from either our courses
-  #   or scripts dashboard db tables.
-  # @param hidden [Boolean] True if the passed in item is hidden
+  # @param script [Course|Script] A row object from our course or script dashboard db table.
   # @return {AssignableInfo} without strings translated
-  def self.assignable_info(course_or_script)
-    name = ScriptConstants.teacher_dashboard_name(course_or_script[:name])
-    first_category = ScriptConstants.categories(course_or_script[:name])[0] ||
-        OTHER_CATEGORY_NAME
-    {
+  def self.assignable_info(course_or_script, full_course: false)
+    info = {
       id: course_or_script[:id],
-      name: name,
+      name: ScriptConstants.teacher_dashboard_name(course_or_script[:name]),
       # Would better be called something like assignable_name
-      script_name: course_or_script[:name],
-      category: first_category,
-      position: ScriptConstants.position_in_category(course_or_script[:name], first_category),
-      category_priority: ScriptConstants.category_priority(first_category),
+      script_name: course_or_script[:name]
     }
+
+    if full_course
```

Comment:
i wanted to call out a few things around this conditional:

1. we always overwrite `info[:category]` in [Course#assignable_info](https://github.com/code-dot-org/code-dot-org/blob/c5effb77f2e390494045dd6be227f68b124d69ee/dashboard/app/models/course.rb#L201), so i don't think it is an issue to ignore that property here.
2. we don't really use `info[:position]` because we sort courses by family/year in [Course#valid_courses_without_cache](https://github.com/code-dot-org/code-dot-org/blob/c5effb77f2e390494045dd6be227f68b124d69ee/dashboard/app/models/course.rb#L243), but i especially want to call this out in case there is any concern with ignoring this property for courses.

i want to make it very obvious that we treat scripts and courses differently, but let me know if you think there are better/alternate solutions here.

Reply:
I don't have a 100% better solution to offer here, but the increase in complexity around the `if full_course` conditionals seems to be adding a fair amount of complexity. would it be possible for this method to do less, and let `Course#assignable_info` fill in the missing parts?